### PR TITLE
Tests for /articles/:id and fix for error response when no article is found

### PR DESCRIPTION
**Pull request by kirinrastogi**
This has tests in it. I followed the convention Jacky laid out.

I needed to use checkArticleProperties, but for only one article. So I moved the code to a function for just a single article, and I use it when checking multiple articles.

This passes all of the Mocha tests laid out.

### CODE REVIEW

File: routes/article.js [hunk 1]
```diff
@@ -110,9 +110,11 @@ router.get('/new', (req, res) => {
 router.get('/:id', (req, res) => {
     Article.findById(req.params.id)
         .then((article) => {
+            if (!article) throw new Error('No articles found')
             res.send(article) 
         })
         .catch((err) => {
+            err = err.toString()
             res.status(500).send({
                 err,
```
Comment (kirinrastogi) [inline routes/article.js]: Should I remove the toString line and add ```err: `${err}` ``` in the response?

Comment (kirinrastogi) [inline routes/article.js]: Should I remove the toString line and add ```err: `${err}` ``` in the response?

Comment (kirinrastogi) [inline routes/article.js]: Should I remove the toString line and add ```err: `${err}` ``` in the response?

**Comment by JackyChiu** [comment]
I'm actually so impressed you fixed the commits, good job 👍

File: test/articles.test.js [hunk 2]
```diff
@@ -33,6 +33,31 @@ describe('/articles Route', () => {
...
+                .get(`/articles/${id}`)
+                .end((err, res) => {
+                    checkSingleArticleProperties(res.body)
+                    res.body._id.should.equal(id)
+                    done()
+                })
+        })
+
+        it('should respond with an error if an article id is not found', (done) => {
+            const id = '5846399f9ba87af2501fb03f'
```
Comment (JackyChiu) [inline test/articles.test.js]: Can we get a more explicit name for this? Ex. `fakeId`

File: test/articles.test.js [hunk 3]
```diff
@@ -33,6 +33,31 @@ describe('/articles Route', () => {
...
+            chai.request(app)
+                .get(`/articles/${id}`)
+                .end((err, res) => {
+                    res.body.should.have.property('err')
+                    res.body.reqParams.id.should.equal(id)
```
Comment (JackyChiu) [inline test/articles.test.js]: I think you should add ` res.body.should.have.property('reqParams')` to be consistant

File: test/articles.test.js [hunk 4]
```diff
@@ -151,32 +176,35 @@ describe('/articles Route', () => {
...
+    article.should.have.property('author')
+    article.should.have.property('title')
+    article.should.have.property('description')
+    article.should.have.property('url')
+    article.should.have.property('urlToImage')
+    article.should.have.property('votes')
+    article.should.have.property('publishedAt')
+    article.should.have.property('keywords')
+    article.keywords.should.be.array
+}
```
Comment (JackyChiu) [inline test/articles.test.js]: Very nice 👌

File: test/articles.test.js [hunk 5]
```diff
@@ -151,32 +176,35 @@ describe('/articles Route', () => {
...
+
 function checkVotesSorted(res) {
     const articles = res.body
     const sorted = articles.every((val, i, arr) => i === 0 || arr[i - 1].votes >= val.votes)
-    sorted.should.be.true
+    sorted.should.to.be.true
```
Comment (JackyChiu) [inline test/articles.test.js]: "Sorted should be true" sound right to me, why the change?

File: test/articles.test.js [hunk 6]
```diff
@@ -151,32 +176,35 @@ describe('/articles Route', () => {
 function checkReponse(res) {
     res.should.have.status(200)
     res.body.should.be.arrary
-    res.body.should.not.be.empty
```
Comment (JackyChiu) [inline test/articles.test.js]: Why remove this?

File: routes/article.js (hunk 1 above)
Comment (JackyChiu) [inline routes/article.js]: What happens if you use use the `Error` object?

Comment (JackyChiu) [inline routes/article.js]: Also can you give me the benefits of user the built in `Error` object vs throwing of a string?

File: test/articles.test.js (hunk 6 above)
Comment (kirinrastogi) [inline test/articles.test.js]: That was an accident that I didn't see.

File: test/articles.test.js (hunk 5 above)
Comment (kirinrastogi) [inline test/articles.test.js]: Mistake from rebasing improperly.

File: routes/article.js (hunk 1 above)
Comment (kirinrastogi) [inline routes/article.js]: I thought it was better practice, but I should just throw a `String` so I don't need to convert an `Error` to a `String`. When I respond with an `Error` object, the `err === {}`

Comment (JackyChiu) [inline routes/article.js]: It could be better practise, I just don't know the benefits of it yet. But it seemed like you were trying to get a `String` in the end anyways so I didn't see the use for the `Error` object

File: test/articles.test.js [hunk 7]
```diff
@@ -198,6 +208,13 @@ function checkSingleArticleProperties(article) {
     article.keywords.should.be.array
 }
 
+function checkValidError(res) {
+    res.should.have.status(500)
+    res.body.should.not.be.empty
+    res.body.should.have.property('err')
+    res.body.should.have.property('reqParams')
```
Comment (JackyChiu) [inline test/articles.test.js]: Just a thought for the future, maybe we should change our `reqQuery` and `reqParams` to be `givens`. This way we have a consistent error message. (Currently we have both due to some routes taking params and some routes taking queries)

**Comment by JackyChiu** [comment]
LGTM 👍

File: test/articles.test.js (hunk 7 above)
Comment (kirinrastogi) [inline test/articles.test.js]: I was thinking the same thing!

Comment (JackyChiu) [inline test/articles.test.js]: I'll create an issue for it
